animal_test.go

```diff
@@ -39,20 +39,20 @@ func TestAskUserYesOrNo(t *testing.T) {
 
 func TestGetUserYesOrNo(t *testing.T) {
 	t.Parallel()
-	var stdin bytes.Buffer
-	stdin.Write([]byte("yes\n"))
+	var input bytes.Buffer
+	input.Write([]byte("yes"))
 	testCases := []struct {
 		question string
 		want     string
 	}{
 		{question: "Is it a horse", want: "yes"},
-		{question: "Is it yes or no", want: "no"},
-		{question: "Does it have 2 legs", want: "no"},
+		{question: "Is it yes or no", want: "yes"},
+		{question: "Does it have 2 legs", want: "yes"},
 	}
 	for _, testCase := range testCases {
-		got := animal.GetUserYesOrNo(stdin)
+		got := animal.GetUserYesOrNo(input.String())
```

Comment:
It's clear that if we're going to use `input` to generate fake input for the test, we'll have to pass it to `GetUserYesOrNo`, isn't it? So that function needs to take two things: the question, as a string, and the input source, as an `io.Reader`. Then the function will read from that instead of `os.Stdin`.

In the test, we'll pass in the fake input contained in the `bytes.Buffer`, but in the 'real' program we'll pass `os.Stdin` instead.

Reply:
Fixed this during the mentoring session.